nailgun/entity_mixins.py

```diff
@@ -82,34 +82,31 @@ def raise_task_timeout():
     timer = threading.Timer(timeout, raise_task_timeout)
 
     # Poll until the task finishes. The timeout prevents an infinite loop.
+    path = '{0}/foreman_tasks/api/tasks/{1}'.format(server_config.url, task_id)
     try:
         timer.start()
-        path = '{0}/foreman_tasks/api/tasks/{1}'.format(
-            server_config.url,
-            task_id
-        )
         while True:
             response = client.get(path, **server_config.get_client_kwargs())
             response.raise_for_status()
             task_info = response.json()
-            if task_info['state'] != 'running':
+            if task_info['state'] in ('paused', 'stopped'):
                 break
             time.sleep(poll_rate)
     except KeyboardInterrupt:
         # raise_task_timeout will raise a KeyboardInterrupt when the timeout
         # expires. Catch the exception and raise TaskTimedOutError
-        raise TaskTimedOutError('Timed out polling task {0}'.format(task_id))
+        raise TaskTimedOutError(
+            'Timed out polling task {0}. Task information: {1}'
+            .format(task_id, task_info)
+        )
```

Comment:
I figure debugging may be easier if we include the entire `task_info` dict. This'll let us answer questions like "what state was the task in when polling timed out?"